src/datanode/src/version_reporter.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use common_meta::version_reporter::{
+    GreptimeVersionReport, Mode as VersionReporterMode, Reporter, VersionReportTask,
+    VERSION_REPORT_INTERVAL, VERSION_UUID_KEY,
+};
+use object_store::{ErrorKind, ObjectStore};
+use servers::Mode;
+
+use crate::datanode::DatanodeOptions;
+
+fn get_uuid_file_name() -> String {
+    format!("./.{}", VERSION_UUID_KEY)
+}
+
+async fn get_uuid(file_name: &str, object_store: ObjectStore) -> Option<String> {
+    let mut uuid = None;
+    let bs = object_store.clone().read(file_name).await;
+    match bs {
+        Ok(bytes) => {
+            let result = String::from_utf8(bytes);
+            if let Ok(s) = result {
+                uuid = Some(s);
+            }
+        }
+        Err(e) => match e.kind() {
+            ErrorKind::NotFound => {
+                let bs = uuid::Uuid::new_v4().to_string().into_bytes();
+                let write_result = object_store.clone().write(file_name, bs.clone()).await;
+                if write_result.is_ok() {
+                    uuid = Some(String::from_utf8(bs).unwrap());
+                }
+            }
+            _ => {}
+        },
+    }
+    return uuid;
+}
+
+pub async fn get_report_task(
+    opts: &DatanodeOptions,
+    object_store: ObjectStore,
+) -> Option<Arc<VersionReportTask>> {
+    let uuid_file_name = get_uuid_file_name();
```

Comment:
`uuid_file_name` could be moved closer to the place where it's used, so it will burn less cognitive resource 😅